pkgs/tools/misc/pls/default.nix

```diff
@@ -0,0 +1,23 @@
+{ lib, python3 }:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "pls";
+  version = "4.0.2";
+
+  src = python3.pkgs.fetchPypi {
+    inherit pname version;
+    sha256 = "0j0kiv5crbm5zbgpaa17755rf3iga78hnf7nbj001jziqkns580z";
+  };
+
+  propagatedBuildInputs = with python3.pkgs; [
+    pyyaml
+    rich
+  ];
+
```

Comment:
I would, but the tests aren't included in the PyPi files. So I can use `fetchFromGitHub` instead and run the tests, but there's no `setup.py` because it's a poetry project. Can I use `mkPoetryApplication` instead of `buildPythonApplication`? Or isn't that recommended/accepted in nixpkgs?
Btw, is `pythonImportsCheck` a logical check for this package? It should really only provide the `pls` binary.

Reply:
> I would, but the tests aren't included in the PyPi files.

Thats not an issue at all and pretty normal to do.

> Can I use `mkPoetryApplication` instead of `buildPythonApplication`?

It should be enough to set ``format = "pyproject"`` and add poetry or better poetry-core to nativeBuildInputs.

> Btw, is `pythonImportsCheck` a logical check for this package? It should really only provide the `pls` binary.

If it can be imported, just add one.

Reply:
Thank you! I didn't find the `format = "pyproject"` in the documentation before, so I didn't know that was possible.